rust-src/wallet_library/src/proofs.rs

```diff
@@ -0,0 +1,134 @@
+use anyhow::Result;
+use concordium_base::{
+    common::base16_decode,
+    id::{constants, types::*},
+    web3id::{OwnedCommitmentInputs, Presentation, Request, Web3IdAttribute, Web3IdSigner},
+};
+use serde::Deserialize as SerdeDeserialize;
+
+/// Serializeable wrapper for a SecretKey.
+#[derive(SerdeDeserialize)]
+pub struct Web3SecretKey(#[serde(deserialize_with = "base16_decode")] ed25519_dalek::SecretKey);
+
+impl Web3IdSigner for Web3SecretKey {
+    fn id(&self) -> ed25519_dalek::VerifyingKey { self.0.id() }
+
+    fn sign(&self, msg: &impl AsRef<[u8]>) -> ed25519_dalek::Signature { self.0.sign(msg) }
+}
+
+/// The input used for creating a web3Id proof. It requires the request itself,
+/// the global context of the chain, and the secret commitments for the
+/// attributes.
+#[derive(SerdeDeserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct Web3IdProofInput {
+    request:           Request<constants::ArCurve, Web3IdAttribute>,
+    global_context:    GlobalContext<constants::ArCurve>,
+    commitment_inputs:
+        Vec<OwnedCommitmentInputs<constants::ArCurve, Web3IdAttribute, Web3SecretKey>>,
+}
+
+/// creates a web3Id proof.
+pub fn create_web3_id_proof(
+    input: Web3IdProofInput,
+) -> Result<Presentation<constants::ArCurve, Web3IdAttribute>> {
+    let presentation = input.request.prove(
+        &input.global_context,
+        input.commitment_inputs.iter().map(Into::into),
+    );
+    presentation.map_err(|e| e.into())
```

Comment:
Why do we map the error here? What's wrong with keeping the error that was already there. You can print it just the same.

Reply:
Hmm, I did it to make it an anyhow::Error, but that is not necessary if I don't use the anyhow::Result.